text_generation/causal_lm/cpp/chat_model_lm.cpp

```diff
@@ -0,0 +1,131 @@
+    ov::InferRequest tokenizer =
+        core.compile_model(std::string{argv[1]} + "/openvino_tokenizer.xml", "CPU").create_infer_request();
+    ov::InferRequest detokenizer =
+        core.compile_model(std::string{argv[1]} + "/openvino_detokenizer.xml", "CPU").create_infer_request();
+    // The model can be compiled for GPU as well
+    ov::InferRequest lm =
+        core.compile_model(std::string{argv[1]} + "/openvino_model.xml", "CPU").create_infer_request();
+
+    int64_t total_positions = 0;
+    int32_t global_beam_idx = 0;
+
+    std::string promt;
+    for (;;) {
+        std::cout << "User promt:\n";
+        std::getline(std::cin, promt);
+        std::cout << "\n";
+
+        if (!promt.compare("Stop!"))
+            break;
+
+        promt = generate_chat_promt_gemma(promt);
+
+        auto [input_ids, new_attention_mask] = tokenize(tokenizer, std::move(promt));
+
+        // Initialize inputs
+        lm.set_tensor("input_ids", input_ids);
+
+        auto attention_mask = lm.get_tensor("attention_mask");
+        // attention to all previous tokens + new tokens
+        ov::Shape mask_shape{1, attention_mask.get_shape().at(1) + new_attention_mask.get_shape().at(1)};
+        attention_mask.set_shape(mask_shape);
+        std::fill_n(attention_mask.data<int64_t>(), ov::shape_size(mask_shape), 1);
+
+        auto position_ids = lm.get_tensor("position_ids");
+        position_ids.set_shape(input_ids.get_shape());
+        auto check = attention_mask.get_shape().at(1);
+        // increment position_ids for every token which is sent to model
+        std::iota(position_ids.data<int64_t>(), position_ids.data<int64_t>() + position_ids.get_size(), total_positions);
+        total_positions += position_ids.get_size();
+
+        lm.set_tensor("beam_idx", ov::Tensor{ov::element::i32, {1}, &global_beam_idx});
+
+        const int64_t* prompt_data = input_ids.data<const int64_t>();
+        Parameters parameters{std::vector<int64_t>{prompt_data, prompt_data + input_ids.get_size()}};
+        GroupBeamSearcher group_beam_searcher{parameters};
+        std::vector<int64_t> next_tokens;
+        std::vector<int32_t> next_beams;
+        for (size_t length_count = 0; length_count < parameters.max_new_tokens; ++length_count) {
+            lm.infer();
+            std::tie(next_tokens, next_beams) = group_beam_searcher.select_next_tokens(lm.get_tensor("logits"));
+            if (next_tokens.empty()) {
+                break;
+            }
+            size_t batch_size = next_tokens.size();
+            // Set pointers
+            lm.set_tensor("input_ids", ov::Tensor{ov::element::i64, {batch_size, 1}, next_tokens.data()});
+            lm.set_tensor("beam_idx", ov::Tensor{ov::element::i32, {batch_size}, next_beams.data()});
+            // Set auxiliary inputs
+            ov::Tensor attention_mask = lm.get_tensor("attention_mask");
+            ov::Shape mask_shape{batch_size, attention_mask.get_shape().at(1) + 1};
```

Comment:
maybe we can define `seq_len_dim_idx = 1` and use:
```suggestion
            ov::Shape mask_shape{batch_size, attention_mask.get_shape().at(seq_len_dim_idx) + 1};
```